### PR TITLE
dd

### PR DESCRIPTION
**Pull request by RelinDmitrii**

### CODE REVIEW

File: app/src/main/java/com/example/calculator/Calculator.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package com.example.calculator;
+
+public class Calculator {
+
+    public long Calculate (String first, String second, String operator){
```
Comment (aimiebell) [inline app/src/main/java/com/example/calculator/Calculator.java:5]: I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Method names should start with a lower case letter (NM_METHOD_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-method-names-should-start-with-a-lower-case-letter-nm-method-naming-convention).
Methods should be verbs, in mixed case with the first letter lowercase, with the first letter of each internal word capitalized.

File: app/src/main/java/com/example/calculator/MainActivity.java [hunk 2]
```diff
@@ -1,14 +1,251 @@
...
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+
+        calculator = new Calculator();
+        initView();
+        initButtons();
+        initLayout();
+        ButtonsSetOnClickListener();
```
Comment (aimiebell) [inline app/src/main/java/com/example/calculator/MainActivity.java:64]: I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Method names should start with a lower case letter (NM_METHOD_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-method-names-should-start-with-a-lower-case-letter-nm-method-naming-convention).
Methods should be verbs, in mixed case with the first letter lowercase, with the first letter of each internal word capitalized.

File: .idea/gradle.xml [hunk 3]
```diff
@@ -1,5 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
```
Comment (happocha) [inline .idea/gradle.xml:1]: папочку .idea я бы не коммитил

File: app/src/main/java/com/example/calculator/Calculator.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+        long a = Long.parseLong(first);
```
Comment (happocha) [inline app/src/main/java/com/example/calculator/Calculator.java:7]: похоже что код можно отформатировать, так как где-то не хватает пробелов

File: app/src/main/java/com/example/calculator/Calculator.java [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+        long b = Long.parseLong(second);
+
+        switch (operator){
+            case"+":
+                return a+b;
```
Comment (happocha) [inline app/src/main/java/com/example/calculator/Calculator.java:12]: подумал бы над именами переменных, а и b кажется не очень понятными

File: app/src/main/java/com/example/calculator/MainActivity.java [hunk 6]
```diff
@@ -1,14 +1,251 @@
...
+        buttonResetOne = findViewById(R.id.buttonResetOne);
+        buttonEqually = findViewById(R.id.buttonEqually);
+    }
+
+
+    public void initLayout() {
+        constraintLayout = findViewById(R.id.constraintLayout);
+    }
+
+    public void ButtonsSetOnClickListener() {
```
Comment (happocha) [inline app/src/main/java/com/example/calculator/MainActivity.java:102]: метод с большой буквы

File: app/src/main/java/com/example/calculator/MainActivity.java [hunk 7]
```diff
@@ -1,14 +1,251 @@
...
+                        startOperation = false;
+                    }
+                    startOperationSin = true;
+                    title.setText(title.getText() + clickNumber);
+                }
+            }
+        }
+    }
+
+    public void calculateAndSet(TextView screenSigns, TextView screenNumbers, TextView screenTitle, String operator) {
```
Comment (happocha) [inline app/src/main/java/com/example/calculator/MainActivity.java:212]: я бы подумал как можно больше логики вынести из Активити. Так как это не совсем ее зона ответственности что-то считать.

File: app/src/main/res/layout-land/activity_main.xml [hunk 8]
```diff
@@ -0,0 +1,323 @@
...
+    tools:context=".MainActivity">
+
+
+    <androidx.constraintlayout.widget.Guideline
+        android:id="@+id/GuideLine"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:orientation="horizontal"
+        app:layout_constraintGuide_percent="0.05
+" />
```
Comment (happocha) [inline app/src/main/res/layout-land/activity_main.xml:17]: верстка уехала

**Review by happocha — APPROVED** [review]

File: app/src/main/java/com/example/calculator/MainActivity.java (hunk 6 above)
Comment (aimiebell) [inline app/src/main/java/com/example/calculator/MainActivity.java:102]: I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Method names should start with a lower case letter (NM_METHOD_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-method-names-should-start-with-a-lower-case-letter-nm-method-naming-convention).
Methods should be verbs, in mixed case with the first letter lowercase, with the first letter of each internal word capitalized.

**Pull request merged**